### ISSUE TITLE
Yet another false positive with invariant_booleans

### ISSUE THREAD

**Issue by Cyp** [dart-lang/linter#2696]
Hard to tell if this is duplicate of any of the many other open issues with `invariant_booleans`.

```dart
void f({required bool a, required bool b}) {
  var c = a;
  if (c) {
    return;
  }
  c = b;
  if (c) {  // <-- False positive here.
    return;
  }
  print('hi');
}
```

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
